spec/lib/open_project/scm/adapters/subversion_adapter_spec.rb

```diff
@@ -0,0 +1,405 @@
+
+  before do
+    allow(adapter).to receive(:config).and_return(config)
+  end
+
+  describe 'client information' do
+    it 'sets the Subversion client command' do
+      expect(adapter.client_command).to eq('svn')
+    end
+
+    context 'with client command from config' do
+      let(:config) { { client_command: '/usr/local/bin/svn' } }
+      it 'overrides the Subversion client command from config' do
+        expect(adapter.client_command).to eq('/usr/local/bin/svn')
+      end
+    end
+
+    shared_examples 'correct client version' do |svn_string, expected_version|
+      it 'should set the correct client version' do
+        expect(adapter)
+          .to receive(:scm_version_from_command_line)
+          .and_return(svn_string)
+
+        expect(adapter.client_version).to eq(expected_version)
+        expect(adapter.client_available).to be true
+        expect(adapter.client_version_string).to eq(expected_version.join('.'))
+      end
+    end
+
+    it_behaves_like 'correct client version', "svn, version 1.6.13 (r1002816)\n", [1, 6, 13]
+    it_behaves_like 'correct client version', "svn, versione 1.6.13 (r1002816)\n", [1, 6, 13]
+    it_behaves_like 'correct client version', "1.6.1\n1.7\n1.8", [1, 6, 1]
+    it_behaves_like 'correct client version', "1.6.2\r\n1.8.1\r\n1.9.1", [1, 6, 2]
+  end
+
+  describe 'invalid repository' do
+    describe '.check_availability!' do
+      it 'should not be available' do
+        expect(Dir.exists?(url)).to be false
+        expect(adapter).not_to be_available
+        expect { adapter.check_availability! }
+          .to raise_error(OpenProject::Scm::Exceptions::ScmUnavailable)
+      end
+
+      it 'should raise a meaningful error if shell output fails' do
+        error_string = <<-ERR.strip_heredoc
+          svn: E215004: Authentication failed and interactive prompting is disabled; see the --force-interactive option
+          svn: E215004: Unable to connect to a repository at URL 'file:///tmp/bar.svn'
+          svn: E215004: No more credentials or we tried too many times.
+          Authentication failed
+        ERR
+
+        allow(adapter).to receive(:popen3)
+          .and_yield(StringIO.new(''), StringIO.new(error_string))
+
+        expect { adapter.check_availability! }
+          .to raise_error(OpenProject::Scm::Exceptions::ScmUnauthorized)
+      end
+    end
+  end
+
+  describe 'repository with authorization' do
+    let(:adapter)  { OpenProject::Scm::Adapters::Subversion.new url, root_url, login, password}
+    let(:login)    { 'whatever@example.org' }
+    let(:svn_cmd)  { adapter.send :build_svn_cmd, ['info'] }
+
+    context 'without password' do
+      let(:password) { nil }
+
+       it 'creates the subversion command' do
+        idx = svn_cmd.index('--username')
+        expect(idx).not_to be_nil
+        expect(svn_cmd[idx + 1]).to eq(login)
+        expect(svn_cmd).not_to include('--password')
+      end
```

Comment:
`end` at 111, 6 is not aligned with `it 'creates the subversion command' do` at 106, 7